src/ecs/rendering/systems.rs

```diff
@@ -38,6 +38,46 @@ impl<'a> System<'a> for RenderSystem {
     ) {
         use std::time::Duration;
 
+        // FIXME issue when running in debug mode:
+        //https://github.com/gfx-rs/gfx/blob/master/src/core/src/pso.rs#L279
+        //Fetch current window size for this rendersystem
+        //Compare with current target size
+        if let Some(cur_window_size) = self.renderer.window_size() {
+            //Window size changed
+            for (name, target) in self.pipe.targets.iter_mut() {
+                if cur_window_size != target.size() {
```

Comment:
`regen_target` creates target bound to window (the one you see).
Main target has a name "" (empty str). Thus only targets named "" must be replaced with new target bound to window. Other targets are created by `TargetBuilder` with `Factory`.
Their size should be changed only if they don't have custom size. See `TargetBuilder`'s `custom_size` field.